hashmap.c

```diff
@@ -345,6 +350,43 @@ int hashmap_iterate(map_t in, PFany f, any_t item) {
     return MAP_OK;
 }
 
+/* Swaps data between two selected elements of hashmap. */
+void _hashmap_swap_items(hashmap_map *m, int pos1, int pos2){
+	char* tmp_key;
+	int tmp_in_use;
+	any_t tmp_data;
+
+	tmp_key = m->data[pos1].key;
+	tmp_in_use = m->data[pos1].in_use;
+	tmp_data = m->data[pos1].data;
+
+	m->data[pos1].key = m->data[pos2].key;
+	m->data[pos1].in_use = m->data[pos2].in_use;
+	m->data[pos1].data = m->data[pos2].data;
+
+	m->data[pos2].key = tmp_key;
+	m->data[pos2].in_use = tmp_in_use;
+	m->data[pos2].data = tmp_data;
+}
+
+/* After removing element from hash map there can be need to move back */
+/* previously pushed elements - pushed moved forward because key colision. */
+void _hashmap_shift_chain_after_remove(hashmap_map *m, char *key, int pos){
+	unsigned int colliding_hash_val;
```

Comment:
The variable `colliding_hash_val` is unused.

Reply:
Thank you for catching it. I've just removed colliding_hash_val with next commit (a8a5769)